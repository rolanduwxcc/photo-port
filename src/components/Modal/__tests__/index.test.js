import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModel = jest.fn();
const currentPhoto = { 
    name: "Panoramic village by sea", 
    category: 'landscape', 
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 3
}

afterEach(cleanup);

describe('Modal is rendering', () => {
    it('renders', () => {
        render(<Modal
            onClose={mockToggleModel}
            currentPhoto={currentPhoto}
        />);
    })

    it('matches snapshot', () => {
        const { asFragment } = render(<Modal onClose={mockToggleModel} currentPhoto={currentPhoto}  />)
        expect(asFragment()).toMatchSnapshot()
    })
})

describe('Click Event', () => {
    it('Calls onClose handler', () => {
        const { getByText } = render(<Modal onClose={mockToggleModel} currentPhoto={currentPhoto} />);
        fireEvent.click(getByText('Close this modal'))

        expect(mockToggleModel).toHaveBeenCalledTimes(1);
    });
})
