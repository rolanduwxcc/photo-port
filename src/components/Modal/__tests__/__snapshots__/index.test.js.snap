// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Panoramic village by sea
      </h3>
      <img
        alt="current category"
        src="3.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
